include/librealsense2/dds/dds-device.cpp

```diff
@@ -340,44 +340,50 @@ size_t dds_device::foreach_sensor( std::function< void( const std::string& name
 }
 
 
-size_t dds_device::foreach_video_profile( size_t sensor_index, std::function< void( const rs2_video_stream& profile, bool def_prof ) > fn )
+size_t
+dds_device::foreach_video_profile( size_t sensor_index,
+                                   std::function< void( const rs2_video_stream & profile, bool def_prof ) > fn ) const
 {
-    size_t i = 0;
-    for (; i < _impl->_sensor_to_video_profiles[sensor_index].num_of_profiles ; ++i )
+    auto const & msg = _impl->_sensor_to_video_profiles.at( int( sensor_index ) );  // may throw!
```

Comment:
Need to catch thrown exception.
If exception is thrown here it means sensor_index is not a video sensor, nothing to do, just return 0.

Handle foreach_motion_profile the same way